DevCycle/SSEConnection.swift

```diff
@@ -11,21 +11,55 @@ import LDSwiftEventSource
 
 public typealias MessageHandler = (String) -> Void
 
-class SSEConnection {
+protocol SSEConnectionProtocol {
+    var connected: Bool { get }
+    func openConnection()
+    func close()
+    func reopen()
+}
+
+class SSEConnection: SSEConnectionProtocol {
     private var connection: EventSource
+    private var url: URL
+    private var handler: MessageHandler
+    
+    var connected: Bool
     
     init(url: URL, eventHandler: @escaping MessageHandler) {
+        self.url = url
+        self.handler = eventHandler
+        self.connected = false
```

Comment:
does the EventSource object contain any property representing connection status that could be used instead of tracking this externally?

Reply:
nope, it's all private